src/Picker/AbstractPickerProvider.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\Picker;
+
+use Contao\BackendUser;
+use Knp\Menu\FactoryInterface;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * Abstract class for picker providers.
+ *
+ * @author Andreas Schempp <https://github.com/aschempp>
+ */
+abstract class AbstractPickerProvider implements PickerProviderInterface
+{
+    /**
+     * @var FactoryInterface
+     */
+    private $menuFactory;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * Constructor.
+     *
+     * @param FactoryInterface      $menuFactory
+     * @param TokenStorageInterface $tokenStorage
+     */
+    public function __construct(FactoryInterface $menuFactory, TokenStorageInterface $tokenStorage)
```

Comment:
I prefer having setter injection for the `TokenStorageInterface` because `getUser()` is absolutely optional for an `AbstractPickerProvider`. I think we should have `setTokenStorage()` and `getTokenStorage()`. The `getUser()` already considers the `null` case anyway :)